src/lamp_py/aws/s3.py

```diff
@@ -191,6 +192,37 @@ def object_metadata(obj: str) -> Dict[str, str]:
         raise error
 
 
+def object_exists(obj: str) -> bool:
+    """
+    check if s3 object exists
+
+    :param obj - expected as 's3://my_bucket/object' or 'my_bucket/object'
+
+    :return: True is object exists, otherwise false
+    """
+    try:
+        s3_client = get_s3_client()
+
+        # trim off leading s3://
+        obj = obj.replace("s3://", "")
+
+        # split into bucket and object name
+        bucket, obj = obj.split("/", 1)
+
+        s3_client.head_object(
+            Bucket=bucket,
+            Key=obj,
+        )
+        return True
+
+    except botocore.exceptions.ClientError as exception:
+        if exception.response["Error"]["Code"] == "404":
+            return False
+        raise exception
```

Comment:
raise all other exceptions, The 404 error code is the only thing that should be returned if the object does not exist. Any other exceptions would be undefined behavior